source/client-side-encryption/tests/README.rst

```diff
@@ -0,0 +1,223 @@
+
+- ``database_name`` and ``collection_name``: The default database and collection
+  to use in operations. This can be overridden by a per operation ``object``.
+
+- ``encryption_setup``: Setup that should occur for each test.
+
+  - ``collections``: Describes collections that should be created and prepopulated.
+
+    - ``database``: The database for the collection to create.
+
+    - ``collection``: The name of the collection to create.
+
+    - ``data``: A list of documents, or a file path, to insert. If this is an encrypted
+      collection (specified by the jsonSchema) the data should be inserted *without*
+      automatic encryption (i.e. create a client without configuring encryption to
+      insert this data).
+
+    - ``remoteSchema``: A document or file path for the schema that should be configured remotely on the server.
+
+    - ``external``: If true, create this collection on the external key vault.
+
+- ``tests``: An array of tests that are to be run independently of each other.
+  Each test will have some or all of the following fields:
+
+  - ``description``: The name of the test.
+
+  - ``skipReason``: Optional, string describing why this test should be
+    skipped.
+
+  - ``clientOptions``: Optional, parameters to pass to MongoClient().
+
+    - ``clientSideEncryptionOptions``
+
+      - ``keyManagementService``:
+
+        - ``name``: Fixed as "aws-kms".
+
+        - ``uri``: Either "real" or "mock".
+
+      - ``mongocryptd``:
+
+        - ``uri``: Either "real" or "mock".
+
+      - ``schemas``: An array of schemas for local configuration.
+
+        - ``database``: The database name for the target collection.
+
+        - ``collection``: The name of the target collection.
+
+        - ``schema``: The schema document.
+
+  - ``kmsPreCmd``: Optional, a command to send to the mock KMS server
+    before the test starts.
+
+  - ``kmsPostCmd``: Optional, a command to send to the mock KMS server
+    after the test ends.
+
+  - ``mockupcryptdPostCmd``: Optional, a command to send to the mock
+    mongocryptd after the test ends.
+
+  - ``operations``: Array of documents, each describing an operation to be
+    executed. Each document has the following fields:
+
+    - ``name``: The name of the operation on ``object``.
+
+    - ``object``: If present, indicates the target collection.
+
+      - ``database``: The database name for the target collection.
+
+      - ``collection``: The name of the target collection.
+
+    - ``collectionOptions``: Optional, parameters to pass to the Collection()
+      used for this operation.
+
+    - ``command_name``: Present only when ``name`` is "runCommand". The name
+      of the command to run. Required for languages that are unable preserve
+      the order keys in the "command" argument when parsing JSON/YAML.
+
+    - ``arguments``: Optional, the names and values of arguments.
+
+    - ``result``: The return value from the operation, if any. This field may
```

Comment:
Since `operations` is plural and contains a list of operation documents, I think this field belongs within each operation. AFAIK, that is what the transactions spec tests already do and it's what I've implemented in my PR to revise the CRUD spec test format in https://github.com/mongodb/specifications/pull/453.

Note that Transactions and CRUD still diverge a bit because CRUD does not represent errors under `result` but instead has a separate `error` field. This is mainly because a BulkWriteException may have a BulkWriteResult in it and we want to assert (a) that error occurred and (b) some state about the result object. I don't think that's relevant in your case, so no objections if you want to continue using the Transactions format here (with `errorContains` and `errorCodeName`) for simplicity.

Reply:
It is within each operation, unless I indented it wrong. Example: https://github.com/mongodb/specifications/pull/447/files#diff-d25b8fce66348003fdf8d126eda5c8ccR76

Reply:
Ah, my mistake. I've been looking at too much RST and YAML recently :P